components/builder-depot/src/server.rs

```diff
@@ -91,7 +88,12 @@ impl TestableBroker {
         let msg_type = &TypeId::of::<M>();
         match self.message_map.get(msg_type) {
             Some(message) => Ok(parse_from_bytes::<R>(message).unwrap()),
-            None => panic!("Unable to find a message of type {:?}", msg_type),
+            None => {
+                match self.error_map.get(msg_type) {
+                    Some(error) => Err(error.clone()),
+                    None => panic!("Unable to find a message of type {:?}", msg_type),
```

Comment:
you are right. I was actually thinking about this last night taking out the garbage and clearly did not think again afterwards!